ntbs-service/Models/Entities/TreatmentEvent.cs

```diff
@@ -20,7 +20,7 @@ public class TreatmentEvent : ModelBase, IHasRootEntityForAuditing
         [Required(ErrorMessage = ValidationMessages.RequiredEnter)]
         [ValidClinicalDate]
         [AssertThat(@"EventDateAfterDob", ErrorMessage = ValidationMessages.DateShouldBeLaterThanDob)]
-        [AssertThat(@"EventDateAfterNotificationDate", ErrorMessage = ValidationMessages.DateShouldBeLaterThanNotification)]
+        [AssertThat(@"EventDateAfterNotificationDate || (IsNotificationPostMortem && TreatmentEventIsDeathEvent)", ErrorMessage = ValidationMessages.DateShouldBeLaterThanNotification)]
```

Comment:
Think I would have preference towards this being its own field/property at this point

Reply:
If you get round to this, might be nice to use nameof(<Field>) rather than the string. I've been fixing it as far as I remember when coming across it in instances where it's easily possible.